### ISSUE TITLE
tensorflow dependency

### ISSUE THREAD

**Issue by msteijaert** [chemprop/chemprop#85]
Why is tensorflow listed as a dependency in `setup.py`? As far as I know, chemprop does not need it as it is fully based on pytorch. Also tensorboard and tensorboardX do not need it.

Is it possible to remove this dependency? This reduces the size of the conda environment (or docker container) also and reduces the risk of version conflicts with other dependencies.

**Comment by swansonk14** [comment]
Hi @msteijaert, tensorflow is listed as a dependency because it is needed to run tensorboard on the log files saved by tensorboardX. Do you know if there's a way to view tensorboard files without having tensorflow installed?

Regardless, I agree that it's worth it to remove the dependency since it's huge and is not a core part of Chemprop. I removed it in this commit and added an explanation in the README for anyone who wants to install it: https://github.com/chemprop/chemprop/commit/d16b4fcb0b7e674e729368896ace3eadf0d82f61.

**Comment by msteijaert** [comment]
Hi @swansonk14, thanks for removing it. I have installed the environment as documented in your README.md and I have installed tensorboard (`pip install tensorboard`) without tensorflow in the same conda environment. With this, I can run tensorboard with `--log_dir` pointing to one of the output dirs created with chemprop training. I do get a warning `TensorFlow installation not found - running with reduced feature set.`, but I'm still able to visualize the log files. As we are using torch + tensorboardx to create the files, I doubt if there is any information missing due to this "reduced feature set".
